elastic/kibana
# [Security Solution][Entity Analytics] Scoping the entity store to spaces

## Summary

This PR introduces Kibana Spaces support for the Entity Store.
It implements https://github.com/elastic/security-team/issues/10530




### How to test

1. Add some host/user data
    * Easiest is to use [elastic/security-data-generator](https://github.com/elastic/security-documents-generator)
2. Make sure to add `entityStoreEnabled` under `xpack.securitySolution.enableExperimental` in your `kibana.dev.yml`
3. Make sure to create a second space other than `default`, either via the UI or the spaces API.
4. In the default space kibana dev tools, call the `POST kbn:/api/entity_store/engines/{entity_type}/init {}` route for either `user` or `host`.
5. Switch to the other space and call `INIT` again.
6. Check that calling the `GET kbn:api/entity_store/engines` route in each space returns only one engine.
7. Check that calling `GET /.kibana*/_search?q=type:entity-engine-status` returns 2 engines, one in each space.
